### PR TITLE
ci: fix do-audit don't report error

### PR DESCRIPTION
**Pull request by yihau**
#### Summary of Changes

Fixes #

### CODE REVIEW

**Comment by yihau** [comment]
will merge this one after #30727

File: ci/do-audit.sh [hunk 1]
```diff
@@ -30,3 +30,7 @@ cargo_audit_ignores=(
   --ignore RUSTSEC-2023-0001
 )
 scripts/cargo-for-all-lock-files.sh audit "${cargo_audit_ignores[@]}" | $dep_tree_filter
+# avoid the pipeline swallows the error
+if [[ "${PIPESTATUS[0]}" -ne 0 ]]; then
+  exit 1
+fi
```
Comment (t-nelson) [inline ci/do-audit.sh:36]:
```suggestion
(
  set +e
  scripts/cargo-for-all-lock-files.sh audit "${cargo_audit_ignores[@]}" | $dep_tree_filter
  # we want the `cargo audit` exit code, not `$dep_filter`'s
  exit ${PIPESTATUS[0]}
)
```

Comment (yihau) [reply ci/do-audit.sh]: I found that I commit a old suggestion. just pushed the latest version!

**Comment by yihau** [comment]
^^^ it failed at the expected place atm 👻

**Comment by yihau** [comment]
I think it looks correct now after I redirect stderr to stdout 👻 

Screenshot 2023-03-16 at 1 21 12 AM

**Comment by t-nelson** [comment]
i guess we actually need to fix the audit now :sweat_smile:

**Comment by yihau** [comment]
@t-nelson sure! The fixed PR is here https://www.github.com/solana-labs/solana/pull/30727 and I'm waiting for it green 🤞

**Comment by yihau** [comment]
btw, we can make docker output prints like a terminal if we want.

**Review by t-nelson — APPROVED** [review]
:shipit:

**Comment by yihau** [comment]
🪖

**Pull request merged**